frontend/src/static/js/v_summary.js

```diff
@@ -385,6 +400,72 @@ window.vSummary = {
       this.filtered = full;
 
       this.sortFiltered();
+
+      if (this.isMergeGroup) {
+        this.mergeGroup();
+      }
+    },
+    mergeGroup() {
+      this.filtered.forEach((group, groupIndex) => {
+        const dateToIndexMap = {};
+        const mergedCommits = [];
+        const mergedFileFormatContribution = {};
+        let mergedVariance = 0;
+        let totalMergedCommits = 0;
+
+        group.forEach((user) => {
+          this.mergeCommits(user, mergedCommits, dateToIndexMap);
+          mergedCommits.sort(window.comparator((ele) => ele.date));
+
+          this.mergeFileFormatContribution(user, mergedFileFormatContribution);
+
+          totalMergedCommits += user.totalCommits;
+          mergedVariance += user.variance;
+        });
+
+        group[0].commits = mergedCommits;
+        group[0].fileFormatContribution = mergedFileFormatContribution;
+        group[0].totalCommits = totalMergedCommits;
+        group[0].variance = mergedVariance;
+
+        // clear all users and add merged group in filtered group
+        this.filtered[groupIndex] = [];
+        const mergedGroup = group[0];
+        this.filtered[groupIndex].push(mergedGroup);
+      });
+    },
+    mergeCommits(user, merged, dateToIndexMap) {
+      // merge commits with the same date
+      user.commits.forEach((commit) => {
+        const {
+          commitResults, date, insertions, deletions,
+        } = commit;
```

Comment:
Can this be put in one line?

Reply:
Putting this in one line will result to eslint error.

Reply:
Is this because of the extra `,` ?

Reply:
No, I think eslint only allows a max of 3 variables in order for it to be in a single line.

<img width="740" alt="Screen Shot 2019-07-29 at 3 12 48 PM" src="https://user-images.githubusercontent.com/32864116/62028790-b4dcc200-b213-11e9-86c7-17edfa8626bb.png">

Reply:
OK, that's fine.